gammapy/modeling/models/spatial.py

```diff
@@ -1376,7 +1377,7 @@ def evaluate_geom(self, geom):
 
         """
         coords = geom.get_coord(frame=self.frame, sparse=True)
-        return self(coords.lon, coords.lat)
+        return self(Angle(coords.lon).wrap_at(180 * u.deg), coords.lat)
```

Comment:
this fix should be applied to the whitin `evaluate` function adding lon=Angle(lon).wrap_at(self.wrap_angle)`